internal/services/recoveryservices/recovery_services_vault_resource.go

```diff
@@ -183,10 +203,37 @@ func resourceRecoveryServicesVault() *pluginsdk.Resource {
 			pluginsdk.ForceNewIfChange("immutability", func(ctx context.Context, old, new, meta interface{}) bool {
 				return old.(string) == string(vaults.ImmutabilityStateLocked)
 			}),
+			resourceRecoveryServicesVaultSoftDeleteCustomizeDiff,
 		),
 	}
 }
 
+func resourceRecoveryServicesVaultSoftDeleteCustomizeDiff(ctx context.Context, diff *pluginsdk.ResourceDiff, v interface{}) error {
+	if diff.Id() == "" {
+		_, new := diff.GetChange("soft_delete_enabled")
+		if new != nil {
```

Comment:
The schema type of the arg is already boolean with default value. Nil check is not necessary here.